### PR TITLE
Added thread_pool and queue_size setting to FIM DB

### PR DESCRIPTION
**Pull request by jotacarma90**
|Related issue|
|---|
|Closes https://github.com/wazuh/wazuh/issues/14056|
|Closes https://github.com/wazuh/wazuh/issues/14021|

## Description
Hello team, this PR implements the new settings related to RSync.
- `thread_pool`: it indicates the number of threads that RSync can use to perform FIM sync.
- `queue_size`: Size of the queue used by agent for sync messages.

We have also modified the unit tests necessary for the correct testing and verification.

## Configuration options

```xml
  <syscheck>
   
   [...]
   
   <synchronization>
      <enabled>yes</enabled>
      <interval>5m</interval>
      <thread_pool>1</thread_pool>
      <queue_size>16384</queue_size>
      <max_eps>10</max_eps>
    </synchronization>
  </syscheck>
```

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X
- [x] Source installation
- [x] Package installation

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [x] AddressSanitizer
- Memory tests for Windows
  - [x] Scan-build report
- Memory tests for macOS
  - [x] Scan-build report
  - [x] Leaks
  - [x] AddressSanitizer

### CODE REVIEW

File: src/unit_tests/Testing/Temporary/LastTest.log [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Start testing: Jul 20 15:02 CEST
```
Comment (antoniomanuelfr) [inline src/unit_tests/Testing/Temporary/LastTest.log:1]: Remove the file

File: src/unit_tests/Testing/Temporary/CTestCostData.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+---
```
Comment (antoniomanuelfr) [inline src/unit_tests/Testing/Temporary/CTestCostData.txt:1]: Remove the file

**Review by antoniomanuelfr — CHANGES_REQUESTED** [review]

File: src/unit_tests/Testing/Temporary/LastTest.log (hunk 1 above)
Comment (jotacarma90) [reply src/unit_tests/Testing/Temporary/LastTest.log]: Done

File: src/unit_tests/Testing/Temporary/CTestCostData.txt (hunk 2 above)
Comment (jotacarma90) [reply src/unit_tests/Testing/Temporary/CTestCostData.txt]: Done

File: src/syscheckd/src/db/include/db.h [hunk 3]
```diff
@@ -40,6 +40,7 @@ extern "C" {
  * @param file_limit Maximum number of files to be monitored.
  * @param value_limit Maximum number of registry values to be monitored.
  * @param sync_registry_enable Flag to enable the registry synchronization.
+ * @param sync_queue_size Number to define the size of the queue to be synchronized.
```
Comment (chemamartinez) [inline src/syscheckd/src/db/include/db.h:43]: `sync_thread_pool` is not in this parameters list

File: src/unit_tests/Testing/Temporary/CTestCostData.txt (hunk 2 above)
Comment (chemamartinez) [reply src/unit_tests/Testing/Temporary/CTestCostData.txt:1]: This file is still uploaded

File: src/unit_tests/Testing/Temporary/LastTest.log (hunk 1 above)
Comment (chemamartinez) [inline src/unit_tests/Testing/Temporary/LastTest.log:1]: Remove this file as well

**Review by chemamartinez — CHANGES_REQUESTED** [review]

**Review by chemamartinez — APPROVED** [review]

**Review by antoniomanuelfr — APPROVED** [review]

**Review by FrancoRivero — APPROVED** [review]
LGTM

**Comment by FrancoRivero** [comment]
Should these settings be in the default configuration?

File: src/unit_tests/Testing/Temporary/CTestCostData.txt (hunk 2 above)
Comment (chemamartinez) [reply src/unit_tests/Testing/Temporary/CTestCostData.txt:1]: This is still needed to be removed.

**Review by chemamartinez — CHANGES_REQUESTED** [review]

**Pull request merged**